includes/services/core/oauth2.php

```diff
@@ -267,6 +267,8 @@ function request( $url, array $params = array() ) {
 				return $res;
 			} elseif ( $raw_response ) {
 				return wp_remote_retrieve_body( $res );
+			} elseif ( '' === wp_remote_retrieve_body( $res ) ) {
```

Comment:
For some API calls the github service returns an empty body ... but there is the potential that other services might do the same

Reply:
I generally try to add the same behavior to all services where possible. So if this is something that could possibly happen with any other service (OAuth1, HTTP Basic), can we add similar handling there?

Reply:
have added this to he oauth1 and http-basic services now